src/libMessage/Messenger.cpp

```diff
@@ -3214,3 +3214,213 @@ bool Messenger::GetConsensusCommitFailure(
+
+    result.set_portno(portNo);
+
+    for (const auto& mb : microBlockInfo)
+    {
+        MicroBlockInfo& res_mb = *result.add_blocknums();
+        res_mb.set_blocknum(mb.first);
+
+        for (uint32_t shard_id : mb.second)
+        {
+            res_mb.add_shards(shard_id);
+        }
+    }
+
+    if (!result.IsInitialized())
+    {
+        LOG_GENERAL(WARNING, "initialization failed.");
+        return false;
+    }
+    return SerializeToArray(result, dest, offset);
+}
+
+bool Messenger::GetLookupGetMicroBlockFromLookup(
+    const vector<unsigned char>& src, unsigned int offset,
+    map<uint64_t, vector<uint32_t>>& microBlockInfo, uint32_t& portNo)
+{
+    LOG_MARKER();
+
+    LookupGetMicroBlockFromLookup result;
+
+    result.ParseFromArray(src.data() + offset, src.size() - offset);
+
+    if (!result.IsInitialized())
+    {
+        LOG_GENERAL(WARNING, "initialization failed.");
+        return false;
+    }
+
+    portNo = result.portno();
+
+    for (const auto& blocknum : result.blocknums())
+    {
+        vector<uint32_t> tempVec;
+        for (const auto& id : blocknum.shards())
+        {
+            tempVec.emplace_back(id);
+        }
+        microBlockInfo.insert(make_pair(blocknum.blocknum(), tempVec));
+    }
+    return true;
+}
+
+bool Messenger::SetLookupSetMicroBlockFromLookup(vector<unsigned char>& dst,
+                                                 unsigned int offset,
+                                                 const vector<MicroBlock>& mbs)
+{
+    LOG_MARKER();
+    LookupSetMicroBlockFromLookup result;
+
+    for (const auto& mb : mbs)
+    {
+        MicroBlockToProtobuf(mb, *result.add_microblocks());
+    }
+
+    if (!result.IsInitialized())
+    {
+        LOG_GENERAL(WARNING, "initialization failed");
+        return false;
+    }
+
+    return SerializeToArray(result, dst, offset);
+}
+
+bool Messenger::GetLookupSetMicroBlockFromLookup(
+    const vector<unsigned char>& src, unsigned int offset,
+    vector<MicroBlock>& mbs)
+{
+    LOG_MARKER();
+    LookupSetMicroBlockFromLookup result;
+
+    result.ParseFromArray(src.data() + offset, src.size() - offset);
+
+    if (!result.IsInitialized())
+    {
+        LOG_GENERAL(WARNING, "initialization failed");
+        return false;
+    }
+
+    for (const auto& res_mb : result.microblocks())
+    {
+        MicroBlock mb;
+
+        ProtobufToMicroBlock(res_mb, mb);
+
+        mbs.emplace_back(mb);
+    }
+
+    return true;
+}
+
+bool Messenger::SetLookupGetTxnsFromLookup(vector<unsigned char>& dst,
+                                           unsigned int offset,
+                                           const vector<TxnHash>& txnhashes,
+                                           uint32_t portNo)
+{
+    LOG_MARKER();
+
+    LookupGetTxnsFromLookup result;
+
+    result.set_portno(portNo);
+
+    for (const auto& txhash : txnhashes)
+    {
+        copy(txhash.asArray().begin(), txhash.asArray().end(),
```

Comment:
For consistency, I propose you copy the way `repeated bytes tranhashes` in `ProtoMicroBlock` is serialized. (I'll update the Protobuf guide for this)